addons/io_hubs_addon/components/definitions/rigid_body.py

```diff
@@ -0,0 +1,168 @@
+        default=False)
+
+    collisionGroup: EnumProperty(
+        name="Collision Group",
+        description="What collision group this object belongs to. This effects what objects will collide with it.",
+        items=[g for g in collision_masks if g[0] != "avatars"],
+        default="objects")
+
+    collisionMask: BoolVectorProperty(
+        name="Collision Mask",
+        description="What collision groups this object will collide with. Note: the other object must also be set to collide with this object's group.",
+        size=5, subtype='LAYER', options={'ANIMATABLE'},
+        default=[value in ["objects", "triggers", "environment"] for (value, _label, _desc) in collision_masks])
+
+    mass: FloatProperty(
+        name="Mass",
+        description="Object's Mass",
+        default=1)
+
+    linearDamping: FloatProperty(
+        name="Linear Damping",
+        description="Amount of linear damping",
+        default=0,
+        min=0.0,
+        soft_max=1.0,
+    )
+
+    angularDamping: FloatProperty(
+        name="Angular Damping",
+        description="Amount of angular damping",
+        default=0,
+        min=0.0,
+        soft_max=1.0,
+    )
+
+    linearSleepingThreshold: FloatProperty(
+        name="Linear Sleeping Threshold",
+        description="Linear velocity threshold below which the object starts to sleep",
+        default=0.8,
+        min=0.0,
+        soft_max=10.0,
+    )
+
+    angularSleepingThreshold: FloatProperty(
+        name="Angular Sleeping Threshold",
+        description="Angular velocity threshold below which the object starts to sleep",
+        default=1.0,
+        min=0.0,
+        soft_max=10.0,
+    )
+
+    angularFactor: FloatVectorProperty(
+        name="Angular Factor",
+        description="Influence of the object's rotation along the X, Y, and Z axes",
+        size=3,
+        subtype="XYZ",
+        default=(1.0, 1.0, 1.0),
+        min=0.0,
+        soft_max=10.0,
+    )
+
+    gravity: FloatVectorProperty(
+        name="Gravity", description="Object's Gravity",
+        unit="ACCELERATION",
+        subtype="ACCELERATION",
+        default=(0.0, 0.0, -9.8))
+
+    def gather(self, export_settings, object):
+        props = super().gather(export_settings, object)
+        props['collisionMask'] = [value for i, (value, _label, _desc) in enumerate(
+            collision_masks) if self.collisionMask[i]]
+
+        # prefer to store as an array for new components
+        props['angularFactor'] = [
+            self.angularFactor[0],
+            self.angularFactor[2] if export_settings['gltf_yup'] else self.angularFactor[1],
+            self.angularFactor[1] if export_settings['gltf_yup'] else self.angularFactor[2],
+        ]
+        props['gravity'] = [
+            self.gravity[0],
+            self.gravity[2] if export_settings['gltf_yup'] else self.gravity[1],
+            self.gravity[1] if export_settings['gltf_yup'] else self.gravity[2],
+        ]
```

Comment:
Nit: I believe self.angularFactor and self.gravity XYZ values could be specified by .x .y .z instead of by index, which would make things a little clearer.